UI/EnergyLabel.gd

```diff
@@ -0,0 +1,20 @@
+extends Label
+
+var energy_component: EnergyComponent
+
+# Called when the node enters the scene tree for the first time.
+func _ready() -> void:
+	energy_component = PlayerManager.player.get_energy_component()
+	energy_component.on_energy_changed.connect(on_energy_changed)
+	CardManager.on_card_action_finished.connect(on_card_played)
+	update_label()
+
+
+func on_card_played(_card: CardWorld) -> void:
```

Comment:
I was also notice that you have the function with a card parameter that is not used
I understand it is because the `on_card_action_finished` returns card. Instead of having a parameter that you don't use, I suggest doing:
`CardManager.on_card_action_finished.connect(on_card_played.unbind(1))` which will basically ignore the value that is given by `on_card_action_finished`
Because right now I was wondering if `on_card_player` was an override or something like that